File: leetcode/c++/70_climbing_stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int dp[2] {1, 2};
        if (n <= 2) {
            return dp[n - 1];
        }
        for (auto i=3; i<=n; ++i) {
            auto tmp {dp[1]};
            dp[1] += dp[0];
            dp[0] = tmp;
        }
        return dp[1];
    }
};

File: leetcode/rust/src/70_climbing_stairs.rs
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        let mut dp = vec![0,1,2];
        dp[1] = 1;
        dp[2] = 2;
        for i in 3..=n as usize {
            dp.push(dp[i-1] + dp[i-2])
        }
        dp[n as usize]
    }
}